docs/spec.md

```diff
@@ -704,195 +708,375 @@ expressions must not be preceded by whitespace before the `[` character,
-of the operation can be determined from the left-hand side alone.
-Comparison operators `<` `<=` `>` `>=` stand for less, less or equal,
+Binary expressions can only be evaluated if the operands are of the
+same type. For example, you cannot add a string to a number. There is
+no automated type conversion of operands.
+
+There are a variety of binary operators: arithmetic, concatenation,
+logical, and comparison operators.
+
+| Operator            | Operands  | Result   | Description   |
+| ------------------- | --------- | -------- | ------------- |
+| `+` `-` `*` `/` `%` | `num`     | `num`    | arithmetic    |
+| `+`                 | `string`  | `string` | concatenation |
+| `+`                 | array     | array    | concatenation |
+| `and` `or`          | `bool`    | `bool`   | logical       |
+| `<` `<=` `>` `>=`   | `num`     | `bool`   | comparison    |
+| `<` `<=` `>` `>=`   | `string`  | `bool`   | comparison    |
+| `==` `!=`           | all types | `bool`   | comparison    |
+
+### Arithmetic and Concatenation Operators
+
+The _arithmetic operators_ `+`, `-`, `*`, `/`, and `%` stand for addition,
+subtraction, multiplication, division, and the [modulo operator]. The
+symbol `+` can also be used to concatenate strings and arrays.
+
+The _modulo operator_ `%`, also known as the remainder operator, returns the
+remainder of a division operation. For example, `10 % 3` results in
+`1`, because `10` divided by `3` has a remainder of `1`.
+
+The _concatenation operator_ `+`, combines two strings or two arrays
+together. For example, `"fire"` + `"engine"` combines into the string
+`"fireengine"`.
+
+### Logical Operators
+
+The _logical operators_ `and` and `or` are used to perform
+[logical conjunction] and [logical disjunction]. They are used to
+perform logical operations on boolean values with type `bool`.
+
+The `and` operator evaluates to `true` if both operands are `true`. The
+`or` operator evaluates to `true` if either operand is `true`.
+
+These operators perform [short-circuit evaluation], which means that the
+right-hand side of the operator is not evaluated if the result of the
+operation can be determined from the left-hand side alone.
+
+For example, the expression `false and true` evaluates to `false`
+because the first operand is `false`. The second operand does not need
+to be evaluated because the result of the expression is already known
+to be `false`.
+
+### Comparison Operators
+
+The _comparison operators_ `<` `<=` `>` `>=` stand for less, less or equal,
 greater, greater or equal. Their operands may be `num` or `string`
 values. For `string` types [lexicographical comparison] is used.
 
-The unary operator `-` stands for the negative sign and can only be used
-with `num`. The unary operator `!` stands for [logical negation] and
-can only be used with `bool`. Unary operators `-` and `!` must not be
-followed by horizontal whitespace.
+The _comparison operators_ `==` and `!=` compare two operands of the
```

Comment:
I think something should be said about the other comparison operators with regard to strings and how they are ordered (lexicographically?)

Reply:
This here is just about equality and in-equality.
The paragraph above describing `<` `<=` `>` `>=`  mentions lexicographic order - is that not enough?

Reply:
i've added that one can compare even composite types...

Reply:
:facepalm: I don't know what I was reading or what was going through my head. You are 100% correct - it is there in black and white. Sorry for the bogus comment.